### ISSUE TITLE
Conflict with persistgraphql package

### ISSUE THREAD

**Issue by lemccarthy** [Swydo/meteor-graphql#5]
If you have both swydo:graphql and persistgraphql packages installed, swydo will throw an error on the index.js file inside the persistgraphql package, causing the program to not run.

**Comment by jamiter** [comment]
HI @lemccarthy, could you make a reproduction for this in a separate github repo?

I'm not sure what package you referring to. Thanks.

**Comment by jamiter** [comment]
Closing untill more information is provided.

**Comment by lemccarthy** [comment]
Sorry it took me so long to get back to you, to reproduce it here are the steps:


1. Get a meteor project.
2. `meteor add swydo:graphql`
3. `meteor npm install persistgraphql`
4. `meteor`

Even without any additional code, the file structure scanning performed by swydo meteor-graphql causes an error on project startup, as it takes issue with some of the files in the "test" directory of the package.

link to the github of persist-graphql: https://github.com/apollographql/persistgraphql

I was able to solve the issue myself by forking your package and adding a line that makes it not scan the `/node_modules` directory.

**Comment by jamiter** [comment]
@lemccarthy, thanks for the update. I'm trying to reproduce it within the tests of this package.

What is the code you added to fix it?

Thanks for the help.

**Comment by lemccarthy** [comment]
```js
import loader from 'graphql-tag/loader';

export default class GraphQLCompiler {
	processFilesForTarget(files) {
		// Fake webpack context
		// See https://github.com/apollographql/graphql-tag/blob/master/loader.js#L26
		const context = {
			cacheable() {},
		};

		files.forEach(function (file) {
			const path = `${file.getPathInPackage()}.js`;
			if (!path.includes("node_modules")) {
				const content = file.getContentsAsString().trim();
				let data = '';

				if (content) {
					data = loader.call(context, content);
				}

				file.addJavaScript({
					data,
					path,
				});
			}
		});
	}
}
```

That's in compiler.js of your package. Nothing complex, just telling it to ignore the node_modules path.

**Comment by jamiter** [comment]
Ah, I now see your fork with the code.

Although the fix works of course, it's hard to test it in the package itself. It doesn't give me any errors when I add the `persistgraphql` package... I'll implement the fix though!

**Comment by jamiter** [comment]
According to https://github.com/meteor/meteor/issues/6037#issuecomment-173362716, it should never load files from node_modules.

What version of Meteor are you on?

**Comment by lemccarthy** [comment]
Meteor 1.4 and it was still occurring, let me try to trigger it again intentionally, get you an exact error message so you can take a look.

**Comment by lemccarthy** [comment]
```
While processing files with swydo:graphql (for target web.browser):

   /~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/error/syntaxError.js:28:15:
   Syntax Error GraphQL request (1:2) Expected Name, found String "{
   author {
   firstName
   lastName
   }
   }
   "
   
   1: {"{\n  author {\n    firstName\n    lastName\n  }\n}\n":9,"query
   otherQuery {\n  person {\n    firstName\n    lastName\n  }\n}\n":10}
   ^
   
   at syntaxError
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/error/syntaxError.js:28:15)
   at expect
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:969:32)
   at parseName
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:91:15)
   at parseField
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:276:21)
   at parseSelection
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:265:72)
   at many
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:1018:16)
   at parseSelectionSet
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:253:17)
   at parseOperationDefinition
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:174:21)
   at parseDefinition
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:127:12)
   at parseDocument
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:109:22)
   at parse
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:46:10)
   at parseDocument
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql-tag/src/index.js:138:16)
   at gql
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql-tag/src/index.js:171:10)
   at Object.lines.some.module.exports
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql-tag/loader.js:44:18)
   at packages/compile-graphql/compiler.js:18:23
   at Array.forEach (native)
   at GraphQLCompiler.processFilesForTarget
   (packages/compile-graphql/compiler.js:12:11)
   
   
   While processing files with swydo:graphql (for target os.osx.x86_64):

   /~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/error/syntaxError.js:28:15:
   Syntax Error GraphQL request (1:2) Expected Name, found String "{
   author {
   firstName
   lastName
   }
   }
   "
   
   1: {"{\n  author {\n    firstName\n    lastName\n  }\n}\n":9,"query
   otherQuery {\n  person {\n    firstName\n    lastName\n  }\n}\n":10}
   ^
   
   at syntaxError
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/error/syntaxError.js:28:15)
   at expect
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:969:32)
   at parseName
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:91:15)
   at parseField
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:276:21)
   at parseSelection
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:265:72)
   at many
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:1018:16)
   at parseSelectionSet
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:253:17)
   at parseOperationDefinition
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:174:21)
   at parseDefinition
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:127:12)
   at parseDocument
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:109:22)
   at parse
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql/language/parser.js:46:10)
   at parseDocument
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql-tag/src/index.js:138:16)
   at gql
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql-tag/src/index.js:171:10)
   at Object.lines.some.module.exports
   (/~~~~/.meteor/packages/swydo_graphql/.0.1.0.1o8msou++os+web.browser+web.cordova/plugin.compile-graphql.os/npm/node_modules/meteor/compile-graphql/node_modules/graphql-tag/loader.js:44:18)
   at packages/compile-graphql/compiler.js:18:23
   at Array.forEach (native)
   at GraphQLCompiler.processFilesForTarget
   (packages/compile-graphql/compiler.js:12:11)
   
   
=> Your application has errors. Waiting for file change.
```

This is the console output (user directory redacted). I am getting this error by following *exactly* the steps I outlined on meteor 1.5. I am also using OSX if that makes any difference. Try creating a blank meteor project, installing those two packages, then starting it.

**Comment by jamiter** [comment]
I was able to reproduce this (in a separate repo) and reported the issue to Meteor. Although I can simply filter out the files from `node_modules`, this would stop one use case from working: When you actually want to load a graphql file from another package... Probably not something people are doing at the moment, so I'll push the fix.

Hopefully I'll get some more info from MDG soon.

**Comment by lemccarthy** [comment]
Thanks, appreciated

**Comment by jamiter** [comment]
Fixed in version 0.1.1

**Comment by jamiter** [comment]
A better fix, after feedback from MDG, has been implemented in version 0.2.0

I would recommend reading the ticket if you're interested in how error handling works in compilers.

Cheers
